core/src/apps/cardano/helpers/credential_params.py

```diff
@@ -0,0 +1,229 @@
+        ):
+            if not SCHEMA_PAYMENT.match(address_params.address_n):
+                credential_params.is_unusual_path = True
+
+        elif address_type in (
+            CardanoAddressType.BASE_SCRIPT_KEY,
+            CardanoAddressType.BASE_SCRIPT_SCRIPT,
+            CardanoAddressType.POINTER_SCRIPT,
+            CardanoAddressType.ENTERPRISE_SCRIPT,
+        ):
+            credential_params.is_other_warning = True
+
+        elif address_type in (
+            CardanoAddressType.REWARD,
+            CardanoAddressType.REWARD_SCRIPT,
+        ):
+            credential_params.is_reward = True
+
+        else:
+            raise ValueError("Invalid address type")
+
+        return credential_params
+
+    @classmethod
+    def stake_params(
+        cls, address_params: CardanoAddressParametersType
+    ) -> "CredentialParams":
+        address_type = address_params.address_type
+        credential_params = CredentialParams(
+            "stake",
+            address_type,
+            address_params.address_n_staking,
+            address_params.staking_key_hash,
+            address_params.script_staking_hash,
+            address_params.certificate_pointer,
+        )
+
+        if address_type == CardanoAddressType.BASE:
+            if address_params.staking_key_hash:
+                credential_params.is_other_warning = True
+            else:
+                if not SCHEMA_STAKING.match(address_params.address_n_staking):
+                    credential_params.is_unusual_path = True
+                if not _do_base_address_credentials_match(
+                    address_params.address_n,
+                    address_params.address_n_staking,
+                ):
+                    credential_params.is_mismatch = True
+
+        elif address_type == CardanoAddressType.BASE_SCRIPT_KEY:
+            if address_params.address_n_staking and not SCHEMA_STAKING.match(
+                address_params.address_n_staking
+            ):
+                credential_params.is_unusual_path = True
+
+        elif address_type in (
+            CardanoAddressType.POINTER,
+            CardanoAddressType.POINTER_SCRIPT,
+        ):
+            credential_params.is_other_warning = True
+
+        elif address_type == CardanoAddressType.REWARD:
+            if not SCHEMA_STAKING.match(address_params.address_n_staking):
+                credential_params.is_unusual_path = True
+
+        elif address_type in (
+            CardanoAddressType.BASE_KEY_SCRIPT,
+            CardanoAddressType.BASE_SCRIPT_SCRIPT,
+            CardanoAddressType.REWARD_SCRIPT,
+        ):
+            credential_params.is_other_warning = True
+
+        elif address_type in (
+            CardanoAddressType.ENTERPRISE,
+            CardanoAddressType.ENTERPRISE_SCRIPT,
+            CardanoAddressType.BYRON,
+        ):
+            credential_params.is_no_staking = True
+
+        else:
+            raise ValueError("Invalid address type")
```

Comment:
```suggestion
            raise RuntimeError  # we didn't cover all address types
```
(this else branch should not happen as `address_type` is validated by protobuf and mypy)

Reply:
Yes, it's more just in case the list is not exhaustive by programmer mistake.
https://github.com/trezor/trezor-firmware/pull/1772/commits/0027ea9a6487ce466de42c3f916ee761b148487a#diff-1b625f81dd2a96105988fd7dce0f740ed62cab35fc1a695b5553d684e38b8e3bR91